include/boost/random/normal_distribution.hpp

```diff
@@ -221,12 +133,30 @@ struct unit_normal_distribution
             if(x < table_x[i + 1]) return x * sign;
             if(i == 0) return generate_tail(eng) * sign;
             RealType y = RealType(table_y[i]) + uniform_01<RealType>()(eng) * RealType(table_y[i + 1] - table_y[i]);
-            if (y < f(x)) return x * sign;
+            if (table_x[i + 1] >= 1) {
+                // Beyond the inflection point; the diagonal is above f(), so we only need to bother
+                // checking f() for points below the diagonal; points above will be rejected.
+                if ((table_x[i] - table_x[i + 1]) * (y - table_y[i]) < (x - table_x[i]) * (table_y[i] - table_y[i + 1])
+                    and y < f(x)) {
+                    return x * sign;
+                }
+            }
+            else if (table_x[i] <= 1) {
+                // Before the inflection point; the diagonal is below f(), so we can accept anything
+                // below without needing to check f()
+                if ((table_x[i] - table_x[i + 1]) * (y - table_y[i]) < (x - table_x[i]) * (table_y[i] - table_y[i + 1])
+                    or y < f(x)) {
+                    return x * sign;
+                }
+            }
+            // Otherwise we're in the block that includes the inflection point, so no simple rule
+            // like the above: just check f(x)
+            else if (y < f(x)) return x * sign;
         }
     }
     static RealType f(RealType x) {
         using std::exp;
-        return exp(-x*x/2);
+        return exp(-(x*x/2));
```

Comment:
I noticed a very slightly performance improvement (1 CPU clock cycle) in performing that calculation this way.  Any of

exp(-0.5*x*x), exp(x*x/-2) and the above show the same 1 clock cycle improvement over the original.  That said, it's a very minor optimization, but not one that gcc is doing on its own (even with -O3).